### PR TITLE
Added ability to add ExceptionProcessorInterceptor to ServiceCollection

### PR DESCRIPTION
**Pull request by uhfath**
In some cases where there is no possibility to use `DbContextOptionsBuilder` this extension adds an ability to use `IServiceCollection` to add `ExceptionProcessorInterceptor` to a list of services with a service type of `IInterceptor`.
Later this could be used to add interceptors where appropriate using `IServiceProvider`.

For example.
Our project uses this to setup a Db context:
```c#
services
    .AddDbContext<MedServiceDbContext>((srv, opts) => opts
        .AddInterceptors(srv.GetServices<IInterceptor>())
        .UseNpgsql(configuration.GetMedServiceDbConnectionString()))
;
```
And there is no way to alter this behavior or subclass a `MedServiceDbContext` to change it's `OnConfiguring`.
This PR adds a helper method to cases like this without hurting any backward compatibility.

To use it simply call `AddExceptionProcessor` on a `IServiceCollection` like usual (i.e. `services.AddExceptionProcessor` somewhere in `ConfigureServices` or elsewhere where appropriate) and then when configuring a `DbContext` use the code like above.

### CODE REVIEW

**Comment by Giorgi** [comment]
Isn't opts instance of `DbContextOptionsBuilder` ? Why don't you use the non generic `UseExceptionProcessor` method?

**Comment by uhfath** [comment]
> Isn't opts instance of `DbContextOptionsBuilder` ? Why don't you use the non generic `UseExceptionProcessor` method?

Yes. `opts` is indeed of `DbContextOptionsBuilder` type. The issue is that we can't alter this setup sequence. I've just showed it as an example.
The only way in our case is through interceptors which has to be registered beforehand. But this project uses internal visibility for them and that's why I've added a PR. Perhaps someone else might come across same limitation in their projects. It doesn't seem to hurt anything.

**Comment by Giorgi** [comment]
I see. Can you also add tests? Just verify that the correct type gets resolved from the container.

**Comment by uhfath** [comment]
Sure. Added appropriate tests.
But also I had to `InternalsVisibleTo` project setting to all the .csproj files so tests would have access to internal classes, which interceptors currently are.

File: EntityFramework.Exceptions.Tests/DatabaseTests.cs [hunk 1]
```diff
@@ -100,7 +100,7 @@ public virtual async Task NotHandledViolationReThrowsOriginalException()
     public virtual async Task DeleteParentItemThrowsReferenceConstraintException()
     {
         var product = new Product { Name = "AN" };
-        var productPriceHistory = new ProductPriceHistory { Product = product, Price = 15.27m, EffectiveDate = DateTimeOffset.UtcNow.Date.AddDays(-10) };
+        var productPriceHistory = new ProductPriceHistory { Product = product, Price = 15.27m, EffectiveDate = DateTime.UtcNow.Date.AddDays(-10) };
```
Comment (Giorgi) [inline EntityFramework.Exceptions.Tests/DatabaseTests.cs:103]: Why this change?

Comment (uhfath) [reply EntityFramework.Exceptions.Tests/DatabaseTests.cs:103]: Sorry, forgot to mention that.
That's because [of this](https://www.npgsql.org/efcore/release-notes/6.0.html#detailed-notes)
This part gave me an exception during test:
```
    Microsoft.EntityFrameworkCore.DbUpdateException : An error occurred while saving the entity changes. See the inner exception for details.
---- System.InvalidCastException : Cannot write DateTimeOffset with Offset=03:00:00 to PostgreSQL type 'timestamp with time zone', only offset 0 (UTC) is supported. Note that it's not possible to mix DateTimes with different Kinds in an array/range. See the Npgsql.EnableLegacyTimestampBehavior AppContext switch to enable legacy behavior.
```
Not sure why it was converted with offset of 03 though `UtcNow` was used but didn't want to waste time digging it since it wasn't relevant in this case.

**Comment by uhfath** [comment]
Do you have any plans on merging this for the next release?

File: EntityFramework.Exceptions.Tests/DatabaseTests.cs (hunk 1 above)
Comment (uhfath) [reply EntityFramework.Exceptions.Tests/DatabaseTests.cs:103]: Should I change something in this PR?

Comment (Giorgi) [reply EntityFramework.Exceptions.Tests/DatabaseTests.cs:103]: @uhfath No, but you can [make a donation](https://ko-fi.com/giorgi)
